orchestration/src/main/java/com/sap/ai/sdk/orchestration/spring/OrchestrationChatModel.java

```diff
@@ -92,18 +125,44 @@ private OrchestrationPrompt toOrchestrationPrompt(@Nonnull final Prompt prompt)
   @Nonnull
   private static com.sap.ai.sdk.orchestration.Message[] toOrchestrationMessages(
       @Nonnull final List<Message> messages) {
-    final Function<Message, com.sap.ai.sdk.orchestration.Message> mapper =
+    final Function<Message, List<com.sap.ai.sdk.orchestration.Message>> mapper =
         msg ->
             switch (msg.getMessageType()) {
               case SYSTEM:
-                yield new SystemMessage(msg.getText());
+                yield List.of(new SystemMessage(msg.getText()));
               case USER:
-                yield new UserMessage(msg.getText());
+                yield List.of(new UserMessage(msg.getText()));
               case ASSISTANT:
-                yield new AssistantMessage(msg.getText());
+                val springToolCalls =
+                    ((org.springframework.ai.chat.messages.AssistantMessage) msg).getToolCalls();
```

Comment:
(Question/Minor)

Do you (not) anticipate class casting exception?

Reply:
Nope, the switch guarantees the type of message.